Userland/Applications/PixelPaint/LayerListWidget.cpp

```diff
@@ -226,22 +242,25 @@ void LayerListWidget::image_did_remove_layer(size_t layer_index)
         m_gadgets[m_moving_gadget_index.value()].is_moving = false;
         m_moving_gadget_index = {};
     }
-    m_gadgets.remove(layer_index);
-    m_selected_layer_index = 0;
+    // No -1 here since a layer has already been removed.
+    auto gadget_index = m_image->layer_count() - layer_index;
+    m_gadgets.remove(gadget_index);
+    m_selected_gadget_index = to_gadget_index(0);
     relayout_gadgets();
 }
 
 void LayerListWidget::image_did_modify_layer_properties(size_t layer_index)
 {
-    update(m_gadgets[layer_index].rect);
+
```

Comment:
Blank line here is a little weird but not a big deal

Reply:
fixed, thanks